### DISCUSSION TITLE
Multiple Entries for Table Blocks?

### DISCUSSION THREAD

**Discussion by SovietRedOctopi** [Obsidian-TTRPG-Community/fantasy-statblocks#247] [Q&A] [answered]
Is there a way to have a table block have multiple entries? 
I'm trying to use it to list weapons & spells for SWADE

I have headers on the table like: Weapon, Range, Damage, Weight, etc... but when I list multiple entries like a sword and then a bow only the last entry shows up

It looks like this in the codeblock and again only the last entry is populating the table when viewing the character sheet
weapon: [Sword, 1, 2d6+STR, 0, 1, 3]
weapon: [Bow, 15/30/60, 2d6, 1, 1, 3]

I'm new to Obsidian and coding related things, so I'm sorry if this is a basic question but I've been wracking my brain on it and feel like I'm doing something wrong

**Comment by Ogolar** [comment]
@SovietRedOctopi 
You have to enter your weapons in a different way:
weapons:
- name:
  desc:
- name:
  desc:

and so on.

At the Moment there ist a bug, so entrys with identical "desc" informations will not show up in the Stat Block.

**Reply by SovietRedOctopi** [reply]
I believe that is the syntax for populating a Trait block, I am trying to use a Table block for weapons instead and the entries are not identical, just subsequent

I tried the name: / desc: as you suggested but it did not fix the issue

**Reply by booge731** [reply]
You probably have it sorted by now, but I hope this helps someone!

The current output of the statblock is not in the format of a traditional table; however using subsequent name and desc terms will give you blocks of text. I had to play around with this a lot to set up a custom statblock, and found a few limitations which prevent the terms and values from appearing properly. It is important to remember that the structure of the data is vital. Each name term must be preceded by two spaces, a dash and space. The related desc term is preceded by four spaces (no dash).
```
  - name: 
    desc: 
``` 
I was encountering the same issue of only the final item being displayed, and was able to resolve it once I had the formatting correct.

Additionally, most special characters will break the statblock, like an asterisk  or colon. If your entry will contain a special character, try surrounding the entire value in double quotes.

**Comment by wsethbrown** [comment]
Wondering something similar. Want to recreate a true table with rows and columns, but not sure if it's actually possible.

**Reply by valentine195** [reply, marked as answer]
It’s not, but this is a feature I have planned.

**Reply by wsethbrown** [reply]
Excellent! I managed to replicate it using Javascript to create a table, but it's definitely not the most user-friendly solution

EDIT: Spoke too soon, can't quite figure out a solution to handle multiple rows

**Reply by TCArknight** [reply]
@wsethbrown - Just in case, I was able to do this with a table group and have the data set up like this where columns are Name, Damage, Bonus, and Range

```
weapons: 
- [Knife, 1, +2, Melee] 
- [Revolver, 2, +2, Short]
```

**Reply by Ascariel1984** [reply]
Maybe I am doing something wrong, but when I try this with my table group, then it doesn't work. Nothing renders. Is there any additional setting to it?

**Comment by claudermilk** [comment]
I am creating stablocks for HarnMaster: Kethira and ran into the same issue. The weapon definitions look similar to SWADE. I got it working with tables! The frontmatter data looks like this:

`weapons: [[Spear,60,60,7T,d8,d8 + 5P,Couched • BR 40 • Slow 1h],[Roundshield,60,60,1,d6,d6 + 1B,Shield Mod 10 • Deflect 3]]`

I have noted that if you need to do any edits within Obsidian, make sure to do it in Source Mode because Live Preview borks up the strings. Also, for longer strings, it will line wrap which can get a little weird.
